### ISSUE TITLE
RibbonCommandHystrixThreadPoolKeyTests throws compilation error in Eclipse with Oracle JDK 1.8.0_131 on WIndows 10

### ISSUE THREAD

**Issue by msamad** [spring-cloud/spring-cloud-netflix#2106]
Hi,
RibbonCommandHystrixThreadPoolKeyTests is throwing compilation error in eclipse and Spring IDE. Getting error `AbstractRibbonCommand is a raw type. References to generic type AbstractRibbonCommand<LBC,RQ,RS> should be parameterized`

Using
* Eclipse (Also tried Spring Tool Suite IDE)
* Oracle JDK 1.8.0_131
* Windows 10

Kindly see the images.
![ribboncommandhystrixthreadpoolkeytests](https://user-images.githubusercontent.com/24932925/27992994-b1225a58-64f4-11e7-94ae-82d91bd88b24.png)
![ribboncommandhystrixthreadpoolkeytests2](https://user-images.githubusercontent.com/24932925/27992998-b818bad2-64f4-11e7-91cc-6f28b7a293e8.png)

Current code
```java
	public static class TestRibbonCommand extends AbstractRibbonCommand {
		public TestRibbonCommand(String commandKey, ZuulProperties zuulProperties) {
			super(commandKey, null, null, zuulProperties);
		}

		@Override
		protected ClientRequest createRequest() throws Exception {
			return null;
		}
	}
```

Changing to following fixes the compilation error and the test class `RibbonCommandHystrixThreadPoolKeyTests` passes all tests successfully. Can someone please confirm if this is the right thing to do - not sure what those tests are doing, haven't done much digging into them.
```java
	public static class TestRibbonCommand extends AbstractRibbonCommand<RestClient, HttpRequest, HttpResponse> {
		public TestRibbonCommand(String commandKey, ZuulProperties zuulProperties) {
			super(commandKey, null, null, zuulProperties);
		}

		@Override
		protected HttpRequest createRequest() throws Exception {
			return null;
		}
	}
```

Thank you.

**Comment by yongsungyoon** [comment]
Hi. I GUESS that the problem is caused by eclipse java compiler. Eclipse always uses eclipse java compiler(ejc) instead of oracle's javac. The code is compiled without any error on oracle's javac. 
I've posted the PR to fix the problem https://github.com/spring-cloud/spring-cloud-netflix/pull/2107.

**Comment by msamad** [comment]
Great! Thank you @yongsungyoon

**Comment by ryanjbaxter** [comment]
Thanks @yongsungyoon

**Comment by ryanjbaxter** [comment]
Closed via #2107
